Exec/RegTests/EB-FlowPastCylinder/prob.H

```diff
@@ -0,0 +1,144 @@
+#ifndef PROB_H
+#define PROB_H
+
+#include <AMReX_Geometry.H>
+#include <AMReX_FArrayBox.H>
+#include <AMReX_REAL.H>
+#include <AMReX_GpuMemory.H>
+
+#include "mechanism.H"
+
+#include "PeleC.H"
+#include "IndexDefines.H"
+#include "Constants.H"
+#include "PelePhysics.H"
+#include "Tagging.H"
+#include "ProblemSpecificFunctions.H"
+#include "prob_parm.H"
+
+AMREX_GPU_DEVICE
+AMREX_FORCE_INLINE
+void
+pc_initdata(
+  int i,
+  int j,
+  int k,
+  amrex::Array4<amrex::Real> const& state,
+  amrex::GeometryData const& geomdata,
+  ProbParmDevice const& prob_parm)
+{
+  // Geometry
+  const amrex::Real* prob_lo = geomdata.ProbLo();
+  const amrex::Real* dx = geomdata.CellSize();
+
+  const amrex::Real x = prob_lo[0] + (i + 0.5) * dx[0];
+
+  const amrex::Real p = prob_parm.dpdx * x + prob_parm.p;
+  amrex::Real rho = 0.0, eint = 0.0;
+  amrex::Real massfrac[NUM_SPECIES] = {0.0};
+  for (int n = 0; n < NUM_SPECIES; n++) {
+    massfrac[n] = prob_parm.massfrac[n];
+  }
+  auto eos = pele::physics::PhysicsType::eos();
+  eos.PYT2RE(p, massfrac, prob_parm.T, rho, eint);
+
+  state(i, j, k, URHO) = rho;
+  state(i, j, k, UMX) = rho * prob_parm.uavg;
+  state(i, j, k, UMY) = 0.0;
+  state(i, j, k, UMZ) = 0.0;
+  state(i, j, k, UEINT) = rho * eint;
+  state(i, j, k, UEDEN) =
+    rho * (eint + 0.5 * (prob_parm.uavg * prob_parm.uavg));
+  state(i, j, k, UTEMP) = prob_parm.T;
+  for (int n = 0; n < NUM_SPECIES; n++) {
+    state(i, j, k, UFS + n) = rho * prob_parm.massfrac[n];
+  }
+}
+
+AMREX_GPU_DEVICE
+AMREX_FORCE_INLINE
+void
+bcnormal(
+  const amrex::Real x[AMREX_SPACEDIM],
+  const amrex::Real s_inter[NVAR],
+  amrex::Real s_ext[NVAR],
+  const int idir,
+  const int sgn,
+  const amrex::Real /*time*/,
+  amrex::GeometryData const& geomdata,
+  ProbParmDevice const& prob_parm,
+  const amrex::GpuArray<amrex::Real, AMREX_SPACEDIM>& /*turb_fluc*/)
+{
+  if (idir == 0) {
+    amrex::Real rho = 0.0, u = 0.0, v = 0.0, w = 0.0, eint = 0.0, T = 0.0;
+    amrex::Real massfrac[NUM_SPECIES] = {0.0};
+    for (int n = 0; n < NUM_SPECIES; n++) {
+      massfrac[n] = prob_parm.massfrac[n];
+    }
+    auto eos = pele::physics::PhysicsType::eos();
+
+    if (sgn == 1) {
+      const amrex::Real p = prob_parm.dpdx * x[0] + prob_parm.p;
+      T = prob_parm.T;
```

Comment:
Instead of this (effectively a Neumann BC for velocity), set the inflow BC using Dirichlet conditions as following:

`T = prob_parm.T` (as is)
`u = prob_parm.uavg`
`v = 0`
`w = 0` (even in 2D, this variable still exists, so just set to 0)

compute p_inter using eos.RTY2P and the interior rho, T, species

use that p and prob_parm.T to compute rho, eint in the ghost cell with eos.PYT2RE